core/cas-server-core-configuration-api/src/main/java/org/apereo/cas/configuration/DefaultCasConfigurationPropertiesSourceLocator.java

```diff
@@ -101,16 +126,21 @@ private static String buildPatternForConfigurationFileDiscovery(final File confi
     }
 
     private PropertySource<?> loadSettingsByApplicationProfiles(final Environment environment, final File config) {
-        val props = new Properties();
-
         val profiles = getApplicationProfiles(environment);
-        val regex = buildPatternForConfigurationFileDiscovery(config, profiles);
-        val configFiles = scanForConfigurationFilesByPattern(config, regex);
+        val profileConfigFiles = scanForConfigurationFiles(config, profiles);
+
+        return new PropertiesPropertySource("applicationProfilesProperties", getProperties(environment, config, profileConfigFiles));
+    }
 
+    // property files processed in order of cas/application first, profiles next, and profiles are first to last
```

Comment:
It would be much better if you could remove all inline comments, and simply turn them into javadocs.